include/my_aes.h

```diff
@@ -131,6 +131,44 @@ int my_aes_get_size(uint32 source_length, enum my_aes_opmode mode);
 
 my_bool my_aes_needs_iv(my_aes_opmode opmode);
 
+/**
+  Encrypt a buffer using AES. Encrypted buffer will have the same size as
+  original one.
+
+  @param source         [in]  Pointer to data for encryption
+  @param source_length  [in]  Size of encryption data
+  @param dest           [out] Buffer to place encrypted data (same size as src)
+  @param key            [in]  Key to be used for encryption
+  @param key_length     [in]  Length of the key. Will handle keys of any length
+  @param mode           [in]  encryption mode
+  @param iv             [in]  16 bytes initialization vector.
+  @return              size of encrypted data, or negative in case of error
+*/
+
+int my_aes_encrypt_nopad(const unsigned char *source, uint32 source_length,
```

Comment:
In binlog_encryption PR I am introducing the same functionality. The difference is only that I am introducing almost the same thing as MariaDB as I need the "context" feature they introduced. So I will be pushing this into repo and this functionalities will overlap. With what I am pushing the same will be possible by setting NO_PAD flag. However, I think the refactoring will need to be done by one who pushes last